main/src/main/java/cgeo/geocaching/ui/ToggleItem.java

```diff
@@ -0,0 +1,54 @@
+package cgeo.geocaching.ui;
+
+import cgeo.geocaching.R;
+
+import android.view.MenuItem;
+
+import androidx.annotation.DrawableRes;
+import androidx.annotation.StringRes;
+
+
+/**
+ * Encapsulates the mechanism to toggle text and image.
+ *
+ * Supports this for menu-item.
+ *
+ * Class is supposed to be used in Activity to toggle ui-elements.
+ */
+public class ToggleItem {
+    public static ToggleItem followMyLocation = new ToggleItem(R.drawable.ic_menu_mylocation, R.drawable.ic_menu_mylocation_off, -1, -1);
+    public static ToggleItem liveMode = new ToggleItem(R.drawable.ic_menu_sync_enabled, R.drawable.ic_menu_sync_disabled, R.string.map_live_disable, R.string.map_live_enable);
```

Comment:
I suggest creating ToggleItem instances not here centrally but in the activity classes where they are used.

Or is the same instance of ToggleItem used in multiple places and ToggleItem is kind of a "ToggleItemType" class? In that case your choice would be correct imho. I suggest then to convert the class to an enum.

Reply:
> I suggest creating ToggleItem instances not here centrally but in the activity classes where they are used.
> 
> Or is the same instance of ToggleItem used in multiple places and ToggleItem is kind of a "ToggleItemType" class? In that case your choice would be correct imho. I suggest then to convert the class to an enum.

The latter was my intention, so the user of the class does not have to think about which string and icon to use.
I will convert it to an enum.